src/librustc/middle/recursion_limit.rs

```diff
@@ -23,9 +24,16 @@ fn update_limit(krate: &ast::Crate, limit: &Once<usize>, name: Symbol, default:
         }
 
         if let Some(s) = attr.value_str() {
-            if let Some(n) = s.as_str().parse().ok() {
-                limit.set(n);
-                return;
+            match s.as_str().parse() {
+                Ok(n) => {
+                    limit.set(n);
+                    return;
+                }
+                Err(e) if e.kind() == &IntErrorKind::Overflow => {
+                    limit.set(usize::max_value());
```

Comment:
I would prefer to also emit a warning here, saying that the recursion limit is too high and has been capped.

Reply:
Why not just error?

Reply:
Is a recursion limit of more than `65536` realisitic? If no, then error IMHO

Reply:
I think we should always emit an error and for all parse failures: `#![recursion_limit = "qwer"]`, `#![recursion_limit = "999999999999999999999"]` and `#![recursion_limit = ""]` should all be emitting an error with specific wording. The overflow case should provide a structured suggestion for `std::usize::MAX` and the other cases should suggest the default `128`.

Reply:
A hard error would not be strictly backwards compatible. Is this okay?

Reply:
Backwards-compatibility, for things we offer such guarantees, is interpreted by the language team in a "does this *actually* break people in practice?"-way. Feel free to crater this, but I have a feeling that no realistic program should be overflowing `usize::max_value()`.

Reply:
I'll add errors for the different parse failures and add a specific suggestion for overflows. In terms of testing, could somebody point me to a good resource that describes testing for error output.

Thanks for all of the input.